src/restore.ts

```diff
@@ -6,10 +6,17 @@ import * as utils from "./utils/actionUtils";
 
 async function run(): Promise<void> {
     try {
-        if (utils.isGhes()) {
-            utils.logWarning(
-                "Cache action is not supported on GHES. See https://github.com/actions/cache/issues/505 for more details"
-            );
+        if (!cache.isAvailable()) {
```

Comment:
Rename `isAvailable` to `isFeatureAvailable` otherwise it is looks like we are talking about a cache item being available.

Reply:
done